packages/design-systems/src/components/composite/Datagrid/SearchFilter/useCustomFilter.ts

```diff
@@ -162,40 +211,45 @@ export const useCustomFilter = <TData>({
    * This will reset the filterRows data state.
    */
   const clearFilterHandler = useCallback(() => {
-    setFilterRows([newEmptyRow({ index: 0, isChangeable: true })]);
+    const emptyRow = [newEmptyRow({ index: 0, isChangeable: true })];
+    setFilterRows(emptyRow);
     setSelectedJointCondition(undefined);
   }, [newEmptyRow]);
 
+  const applyFilterHandler = useCallback(() => {
+    onChangeHandler(filterRows);
+    setNumberOfSearchConditions(calcNumberOfSearchConditions());
+    setPrevFilterRows(filterRows);
+  }, [filterRows, onChangeHandler, calcNumberOfSearchConditions]);
   /**
    * This will add new item to filterRows data state.
    */
   const addNewFilterRowHandler = useCallback(() => {
     setFilterRows((oldState) => {
       const newState = [...oldState];
+      const selectedJointCondition = oldState.find((state) => {
+        return state.currentState.jointCondition === "and" || "or";
```

Comment:
このロジックだと常にtrueが返る(フォールバックされる `"or"` がtrue判定になってしまう)と思うんですが、問題ないんですかね? 

なんとなくいかが正しい気がするんですが。

```suggestion
        return state.currentState.jointCondition === "and" || state.currentState.jointCondition === "or";
```

Reply:
確かに。
型がちゃんとしてる前提で書くなら↓でも。
```ts
      const selectedJointCondition = oldState.find(({ currentState }) => currentState.jointCondition)?.currentState.jointCondition;
```

Reply:
修正しました！